apps/omg_watcher_info/test/omg_watcher_info/order_fee_fetcher_test.exs

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.OrderFeeFetcherTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
```

Comment:
good point. I think this is the TestServer - FakeServer dependency limitation because it changes the URL with each request - making Application.put_env/3 calls. We should step away from this dependency and write our own trivial webserver:
```
  defp start(port) do
    {:ok, sock} = :gen_tcp.listen(port, [{:active, false}])
    spawn(fn -> loop(sock) end)

    receive do
      :stop ->
        :gen_tcp.close(sock)
    end
  end

  defp loop(sock) do
    case :gen_tcp.accept(sock) do
      {:ok, conn} ->
        handler = spawn(fn -> handle(conn) end)
        :gen_tcp.controlling_process(conn, handler)
        loop(sock)

      _ ->
        :ok
    end
  end

  defp handle(conn) do
    body = Jason.encode!(@exchanger_body)
    :ok = :gen_tcp.send(conn, ["HTTP/1.0 ", Integer.to_charlist(200), "\r\n", [], "\r\n", body])
    :gen_tcp.close(conn)
  end
```
you can add ID to requests/responses and have it serve the whole test suite concurrently

Reply:
Talked to @mederic-p a bit. The ID handling on TestServer side is quite straightforward, but injecting the ID into the feature code can add quite a bit of complexity. 

I'm thinking this test might not worth adding the complexity to make it async just yet?